addons/hr_expense/models/hr_expense.py

```diff
@@ -148,6 +149,9 @@ def _compute_product_has_cost(self):
             expense.product_has_cost = expense.product_id and (float_compare(expense.product_id.standard_price, 0.0, precision_digits=2) != 0)
             tax_ids = expense.product_id.supplier_taxes_id.filtered(lambda tax: tax.company_id == expense.company_id)
             expense.product_has_tax = bool(tax_ids)
+            if not expense.product_has_cost and expense.state == 'draft':
+                expense.unit_amount = expense.total_amount_company
+                expense.quantity = 1
```

Comment:
I don't get this :thinking:

Reply:
used when changing the product to recompute properly the fields that will disappear/reappear on the view

Reply:
But we're changing the quantity ? Will it not seem weird to the user ? :thinking:

Reply:
The field is hidden when product_has_cost is True
As we now start to input the expense price from the total_amount not from a product

E.G. your expense for gas is usually at a set price for reimbursment so you just input the quantity
but hotel or restaurant fees depends on the hotel/restaurant price and cannot be predicted. So we can record their prices directly in the total (and the quantity is set to 1 as it isn't really that important)

Reply:
Had a deeper look; got it :)